core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -34,6 +37,8 @@ public class CommitHandler {
   private final TransactionTableMetadataManager tableMetadataManager;
   private final ParallelExecutor parallelExecutor;
 
+  @LazyInit @Nullable private BeforePreparationSnapshotHook beforePreparationSnapshotHook;
```

Comment:
It's a minor thing, but I thought `@LazyInit` also implied that the field was nullable. Should we add `@Nullable` alongside `@LazyInit` as well?

Reply:
> I thought `@LazyInit` also implied that the field was nullable.

Does it point that the field can be null until it's lazily initialized? If so, I think only `@LazyInit` may be enough. `@Nullable` is added since the field can be permanently null if no snapshot hook is set. Does it answer your question?

Reply:
That makes sense. Thanks!